### DISCUSSION TITLE
Does Navidrome support compilations of mp4 tag types (e.g. m4a format)?

### DISCUSSION THREAD

**Discussion by tooandy** [navidrome/navidrome#3407] [General]
As I know, the tag named `tcmp` corresponds to the mp3 file, `compilation` corresponds to the flac file, and `wm/iscompilation` corresponds to the wav file. 
However, the collection tag `cpil` in m4a format (which belongs to MP4) does not seem to be recognized by Navidrome.  This will result in songs that belong to a compilation being split into multiple albums based on the song's artist. 

i find the code here, it has not `cpil` tag. [metadata.go](https://github.com/navidrome/navidrome/blob/3713032f57eea19ffb2ce96bacb497e471b8f395/scanner/metadata/metadata.go#L142)
```
func (t Tags) Compilation() bool           { return t.getBool("tcmp", "compilation", "wm/iscompilation") }]
```

**Comment by deluan** [comment]
Yes, it works with m4a because both ffmpeg and TagLib normalizes tags to common name, and in the case of `cpil`, it becomes `compilation`

**Comment by tooandy** [comment]
OK，Thank you for your answer



---Original---
From: "Deluan ***@***.***&gt;
Date: Fri, Oct 18, 2024 20:20 PM
To: ***@***.***&gt;;
Cc: "Andre ***@***.******@***.***&gt;;
Subject: Re: [navidrome/navidrome] Does Navidrome support compilations of mp4tag types (e.g. m4a format)? (Discussion #3407)




 
Yes, it works with m4a because both ffmpeg and TagLib normalizes tags to common name, and in the case of cpil, it becomes compilation
